internal/gojsonschema/schema.go

```diff
@@ -151,82 +145,44 @@ func (d *Schema) parseSchema(documentNode interface{}, currentSchema *SubSchema)
 	}
 
 	// definitions
-	if existsMapKey(m, KeyDefinitions) {
-		if isKind(m[KeyDefinitions], reflect.Map, reflect.Bool) {
-			for _, dv := range m[KeyDefinitions].(map[string]interface{}) {
-				if isKind(dv, reflect.Map, reflect.Bool) {
-
+	if v, ok := m[KeyDefinitions]; ok {
+		switch mt := v.(type) {
```

Comment:
Previously the code would simply panic if `m[KeyDefinitions]` were a boolean. I've removed it here as we immediately convert the object to a `map[string]interface{}`, which does not seem to be behavior that the original author intended.